streams/src/test/java/org/apache/kafka/streams/integration/AbstractResetIntegrationTest.java

```diff
@@ -269,11 +270,9 @@ public void uncaughtException(Thread t, Throwable e) {
 
         // RESET
         streams = new KafkaStreams(setupTopologyWithIntermediateUserTopic(OUTPUT_TOPIC_2_RERUN), streamsConfiguration);
-        final KafkaStreams handlerReference2 = streams;
         streams.setUncaughtExceptionHandler(new Thread.UncaughtExceptionHandler() {
             @Override
             public void uncaughtException(Thread t, Throwable e) {
-                handlerReference2.close(10, TimeUnit.SECONDS);
```

Comment:
Why remove this? And if we want to remove it, why not also above?

Reply:
I feel this exception handler to close the instance is not necessary as we may just want to let the unexpected exception to be thrown directly than logging it since in the latter case the test will fail but the root cause may be hidden without log4j entries enabled.

I'd remove both of these two exception handlers if that works for you.

Reply:
But without exception handling the exception would be swallowed anyway? So if you are afraid it might get lost, we should rather introduce a global exception variable and set it within the handler. Thus we can rethrow the exception from the main code. WDYT?

Reply:
I think without the exception handling the exception stack trace should be printed to syserr before thread calling `completeShutdown`?

Reply:
I am not sure about this. If you are sure that's the case, we can remove the handler.